types/web3/providers.d.ts

```diff
@@ -0,0 +1,64 @@
+interface JsonRPCRequest {
```

Comment:
@Levino -- Any harm in exporting the interfaces `JsonRPCRequest` and `JsonRPCResponse`? I'd like to have the typings for `payload` as well as `response` for this method -- 
```
web3.currentProvider.send(payload, (err, response) => {}))
```

Also, another thing I've noticed is that currently we don't have any interface defined for Event/Log Filters. Worth defining an interface called `FilterOptions` and use it across maybe? --
```
export interface FilterOptions {
    fromBlock?: BlockType;
    toBlock?: BlockType;
    address: string;
    topics?: Array<string | string[]>;
}
```

Let me know if you'd like me to create a PR for these two and update them. I'd be more than happy to. Thanks :)

Reply:
I just copied the existing type definitions I found at the the `web3` repos source code. Feel free to make a PR for the improvements you wish to have.

Reply:
@Levino -- Sure. Will soon create a PR for the above changes. Thanks!